internal/app/machined/pkg/system/services/etcd.go

```diff
@@ -439,7 +424,143 @@ func (e *Etcd) args(r runtime.Runtime) ([]string, error) {
 		blackListArgs.Set("advertise-client-urls", fmt.Sprintf("https://%s:2379", ips[0].String()))
 	}
 
-	return blackListArgs.Merge(extraArgs).Args(), nil
+	e.args = blackListArgs.Merge(extraArgs).Args()
+
+	return nil
+}
+
+func (e *Etcd) setup(ctx context.Context, r runtime.Runtime, errCh chan error) {
+	var err error
```

Comment:
this can be simplified by wrapping a function which returns error:

```go
  errCh <- func() error {
    if ...; err != nil {
       return err
    }
   
   return nil
  }
}
```

this way we can't miss pushing value to `errCh`

Reply:
Nice :+1: